src/Test/Utilities/Portable/CoreCLRRuntimeEnvironment.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CodeGen;
+using Roslyn.Test.Utilities;
+
+namespace Microsoft.CodeAnalysis.Test.Utilities
+{
+    public class CoreCLRRuntimeEnvironment : IRuntimeEnvironment, IInternalRuntimeUtility
+    {
+        private IEnumerable<ModuleData> _additionalDependencies;
+        private CompilationTestData _testData = new CompilationTestData();
+        private ImmutableArray<byte> _mainImage;
```

Comment:
It feels like these should be fields of some `EmitResult` class, and then we should return that on `Emit`.